internal/plugin/connectors/tcp/mssql/connection_details.go

```diff
@@ -8,36 +8,44 @@ import (
 // ConnectionDetails stores the connection info to the real backend database.
 // These values are pulled from the SingleUseConnector credentials config
 type ConnectionDetails struct {
-	Host     string
-	Port     uint
-	Username string
-	Password string
-	SSLMode  string
+	Host       string
+	Port       uint
+	Username   string
+	Password   string
+	SSLMode    string
+	SSLOptions map[string]string
+}
+
+var sslOptions = []string{
+	"sslrootcert",
+	"sslkey",
+	"sslcert",
 }
 
 const defaultMSSQLPort = uint(1433)
+const defaultSSLMode = "disable"
 
 // NewConnectionDetails is a constructor of ConnectionDetails structure from a
 // map of credentials.
 func NewConnectionDetails(credentials map[string][]byte) *ConnectionDetails {
 
 	connDetails := &ConnectionDetails{}
 
-	if host := credentials["host"]; host != nil {
+	if len(credentials["host"]) > 0 {
```

Comment:
This new type of check (`len`) you have in here will panic if "host" is not defined in the config file I think